test/github_pull_request_test.go

```diff
@@ -0,0 +1,411 @@
+			),
+		}
+
+		r, err := recorder.NewWithOptions(&recorder.Options{
+			CassetteName:  filepath.Join("fixtures", "github", t.Name()),
+			Mode:          recorder.ModeReplayOnly,
+			RealTransport: oauth2Transport,
+		})
+		assert.NoError(t, err, "error creating HTTP recorder")
+
+		// oauth2.Transport mutates the request after go-vcr saves it, so the Authorization header should never be persisted.
+		// However, this is a safeguard in case that implementation changes.
+		r.AddHook(func(interaction *cassette.Interaction) error {
+			delete(interaction.Request.Headers, "Authorization")
+
+			return nil
+		}, recorder.AfterCaptureHook)
+
+		return r
+	}
+
+	stopRecorder := func(t *testing.T, r *recorder.Recorder) {
+		t.Helper()
+
+		if err := r.Stop(); err != nil {
+			t.Log("error stopping recorder:", err)
+		}
+	}
+
+	createGitRepoAtHead := func(t *testing.T, commit string, remotes ...string) *git.Repository {
+		t.Helper()
+
+		cfg := gitconfig.NewConfig()
+		if len(remotes) == 0 {
+			remotes = append(remotes, githubPrFixtureRepository)
+		}
+
+		cfg.Remotes[git.DefaultRemoteName] = &gitconfig.RemoteConfig{
+			Name: git.DefaultRemoteName,
+			URLs: remotes,
+		}
+
+		store := memory.NewStorage()
+		err := store.SetConfig(cfg)
+		assert.NoError(t, err, "error configuring in-memory repository")
+
+		headRef := plumbing.NewReferenceFromStrings(plumbing.HEAD.String(), commit)
+		err = store.SetReference(headRef)
+		assert.NoError(t, err, "error setting reference on in-memory repository")
+
+		return &git.Repository{
+			Storer: store,
+		}
+	}
+
+	makeCtx := func(r *recorder.Recorder, repo *git.Repository) context.Context {
+		ctx := context.WithValue(context.Background(), github.GitHubHttpClientCtxKey, r.GetDefaultClient())
+
+		return context.WithValue(ctx, ghpratt.GitRepositoryOverrideCtxKey, repo)
+	}
+
+	parseTime := func(t *testing.T, timestamp string) time.Time {
+		t.Helper()
+
+		parsed, err := time.Parse(time.RFC3339, timestamp)
+		assert.NoError(t, err)
+
+		return parsed
+	}
+
+	runAttestationCmd := func(t *testing.T, ctx context.Context, artifact *containerImage) {
+		t.Helper()
+
+		ghprCmd := cmd.GitHubPullRequest{}
+		flags := makeGlobalFlags(artifact.digest.String())
+		err := ghprCmd.Init(ctx, flags)
+		assert.NoError(t, err, "failure initializing test command")
+		assert.NoError(t, ghprCmd.Run(), "error running test command")
+	}
+
+	type testCase struct {
+		commit string
+		name   string
+		assert func(t *testing.T, artifact *containerImage, attestation prAttestation)
+	}
+
+	testCases := []*testCase{
+		{
+			name:   "approved pull request with one review",
+			commit: commitWithApproval,
+			assert: func(t *testing.T, artifact *containerImage, attestation prAttestation) {
+				assert.Len(t, attestation.Subject, 2, "attestation should have 2 subjects")
+				// first subject is the commit created from the pull request
+				assert.Equal(t, attestation.Subject[0].Name, fmt.Sprintf("git+%s.git", githubPrFixtureRepository))
```

Comment:
Is the ordering of the subjects list guaranteed?

Reply:
It should be -- they're inserted in the same order in the code and the ordering shouldn't be changed by JSON serialization.

Reply:
Changed this to the `assert.Contains`, so the order shouldn't order regardless